### PR TITLE
Adding background image to widget works.

### PR DESCRIPTION
**Pull request by wszczepanski97**
[Problem] User can't set a background image - it doesn't show in editor mode.
[Solution] Add function setImageSource to read file from user and set as an attribute to image.

Signed-off-by: Wojciech Szczepanski <w.szczepansk@samsung.com>

### CODE REVIEW

File: design-editor/src/utils/path-utils.js [hunk 1]
```diff
@@ -60,6 +61,53 @@ export default {
...
+
+					utils.checkGlobalContext('writeFile') (
+						writePath,
+						event.target.result, {
+							encoding: 'binary'
+						},
+						() => {
+							if (changedAttribute === 'src') {
+								AppManager.getActiveDesignEditor()
+									.getModel().updateAttribute(
```
Comment (hsiwkin) [inline design-editor/src/utils/path-utils.js]: I'd move `AppManager.getActiveDesignEditor().getModel()` outside if (just before it) and assign to a variable as it repeats both in `if` and in `else` block. It should be more readable this way : )

File: design-editor/src/utils/path-utils.js [hunk 2]
```diff
@@ -60,6 +61,53 @@ export default {
...
+
+	/**
+	 * Set a source to background image or image
+	 * @param {Event} event
+	 * @param {string} changedAttribute name of changed attribute
+	 * @param {string} elementId id of changed element
+	 * @param {function} changeImageInfo function that changed info about element in sidebar
+	 */
+	setImageSource(event, changedAttribute, elementId, changeImageInfo) {
+		[].slice.call(event.target.files).forEach(file => {
```
Comment (hsiwkin) [inline design-editor/src/utils/path-utils.js]: we should probably move it to another utils file (eg attribute-utils.js) as it does not exactly fit path-utils.

File: design-editor/src/panel/property/attribute/attribute-element-image.js [hunk 3]
```diff
@@ -169,31 +168,7 @@ class AttributeImage extends DressElement {
...
-							AppManager.getActiveDesignEditor()
-								.getModel().updateAttribute(self._selectedElementId, 'src', readpath);
-							self._updateImageSourcePath(readpath);
-						}
-					);
-				}
-			});
-			reader.readAsBinaryString(file);
-		});
+		pathUtils.setImageSource(event, 'src', this._selectedElementId, this._updateImageSourcePath.bind(this));
```
Comment (hsiwkin) [inline design-editor/src/panel/property/attribute/attribute-element-image.js]: this line is quite long, maybe we should split it into separate lines like this:
```
pathUtils.setImageSource(
   event, 
   'src', 
   this._selectedElementId, 
   this._updateImageSourcePath.bind(this)
);
```

**Comment by singa2000** [comment]
@hsiwkin 
Would you review the patch, please?

**Comment by hsiwkin** [comment]
BTW I've updated this branch with newest commits from master

**Pull request merged**